src/qt/menuitem.cpp

```diff
@@ -183,7 +183,7 @@ void wxMenuItem::ClearExtraAccels()
 
 wxQtAction::wxQtAction( wxMenu *handler, int id, const wxString &text, const wxString &help,
         wxItemKind kind, wxMenu *subMenu, wxMenuItem *menuItem )
-    : QAction( wxQtConvertString( text ), handler->GetHandle() ),
+    : QAction( wxQtConvertString( text ), handler != nullptr ? handler->GetHandle() : nullptr ),
       wxQtSignalHandler( handler ),
       m_mitemId(menuItem->GetId()), m_isCheckable(menuItem->IsCheckable())
```

Comment:
Are you sure this fixes the crash you're seeing? because passing `nullptr` to `wxQtSignalHandler` will also cause a crash when `EmitEvent()` is called: https://github.com/wxWidgets/wxWidgets/blob/5dac42edc07f6e504ecc9b5a9cbb1c559ca9319c/include/wx/qt/private/winevent.h#L38-L45

Do have some snippet reproducing the crash by chance ?

BTW, thanks a lot for working on this and making it working.

Reply:
Wait! maybe `EmitEvent()` mentioned above is the **culprit**!! because the comment there says:
>     // We're only called with the objects of class (or derived from) 
>     // wxWindow, so the cast is safe.

which is not true if the handler is wxMenu, because it derives from `wxEvtHandler` and not from `wxWindow`?

We should investigate...

Reply:
It was crashing when constructing wxMenuItem with nullptr parent, for example:

https://gitlab.com/kicad/code/kicad/-/blob/d0236ca7511b24ef106297482e7c58e65d63d186/common/file_history.cpp#L150-165

I didn't see any other related crashes at runtime.

Reply:
Indeed, constructing `wxMenuItem` with `nullptr` parent is perfectly legal, and is already documented here:
https://docs.wxwidgets.org/3.2/classwx_menu_item.html#a6e9b0e1b786fa84250a42c88d84aed2b:~:text=Menu%20that%20the%20menu%20item%20belongs%20to.%20Can%20be%20NULL%20if%20the%20item%20is%20going%20to%20be%20added%20to%20the%20menu%20later.

And I think we should properly handle this in `wxQt`. i.e.:
- Allow passing `nullptr` to `wxQtAction`.
- Assign a valid parent to `wxQtAction` when the `wxMenuItem` is added or appened to a `wxMenu`.
- Fix `EmitEvent()` to not expect `wxWindow` objects only.

Just let me know if you plan to do this here or in a separate PR. because otherwise, this commit simply hides a potential bug IMHO.

TIA.

Reply:
>Just let me know if you plan to do this here or in a separate PR.

I'll probably leave it to others, since I don't know all the details here.

Reply:
I'll fix it then. just don't include this commit when you finish working on this.

Reply:
See commit [a021a01](https://github.com/wxWidgets/wxWidgets/pull/24210/commits/a021a01c828198dd192372af8de3fdf15a386e4d) in PR #24210

Reply:
>just don't include this commit when you finish working on this.

Commit dropped.